server/streamable_http.go

```
@@ -0,0 +1,810 @@
+		// Start an SSE stream for this request
+		s.handleSSEResponse(w, r, ctx, response, session)
+	} else {
+		// Send a direct JSON response
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusOK)
+		if response != nil {
+			json.NewEncoder(w).Encode(response)
+		}
+	}
+}
+func (s *StreamableHTTPServer) handleSSEResponse(w http.ResponseWriter, r *http.Request, ctx context.Context, initialResponse mcp.JSONRPCMessage, session SessionWithTools) {
```

golangci-lint (1.64.8)

SA4009: argument ctx is overwritten before first use

(staticcheck)